packages/capabilities/src/web3.storage/blob.js

```diff
@@ -0,0 +1,109 @@
+import { capability, Schema, Link, ok, fail } from '@ucanto/validator'
+import { content } from '../blob.js'
+import {
+  equalBlob,
+  equalWith,
+  SpaceDID,
+  and,
+  equal,
+  checkLink,
+} from '../utils.js'
+
+/**
+ * Service capabilities for Blob protocol
+ */
+/**
+ * Capability can only be delegated (but not invoked) allowing audience to
+ * derived any `web3.storage/blob/` prefixed capability for the (memory) space identified
+ * by DID in the `with` field.
+ */
+export const blob = capability({
+  can: 'web3.storage/blob/*',
+  /**
+   * DID of the (memory) space where Blob is intended to
+   * be stored.
+   */
+  with: SpaceDID,
+  derives: equalWith,
+})
+
+/**
+ * `web3.storage/blob//allocate` capability can be invoked to create a memory
+ * address where blob content can be written via HTTP PUT request.
+ */
+export const allocate = capability({
+  can: 'web3.storage/blob/allocate',
+  /**
+   * Provider DID.
+   */
+  with: Schema.did(),
+  nb: Schema.struct({
+    /**
+     * Blob to allocate on the space.
+     */
+    blob: content,
+    /**
+     * The Link for an Add Blob task, that caused an allocation
+     */
+    cause: Link,
+    /**
+     * DID of the user space where allocation takes place
+     */
+    space: SpaceDID,
+  }),
+  derives: (claim, from) => {
+    return (
+      and(equalWith(claim, from)) ||
+      and(equalBlob(claim, from)) ||
+      and(checkLink(claim.nb.cause, from.nb.cause, 'cause')) ||
+      and(equal(claim.nb.space, from.nb.space, 'space')) ||
+      ok({})
+    )
+  },
+})
+
+/**
+ * `blob/accept` capability invocation should either succeed when content is
+ * delivered on allocated address or fail if no content is allocation expires
+ * without content being delivered.
+ */
+export const accept = capability({
+  can: 'web3.storage/blob/accept',
+  /**
+   * Provider DID.
+   */
+  with: Schema.did(),
+  nb: Schema.struct({
+    /**
+     * Blob to accept.
+     */
+    blob: content,
+    /**
+     * Expiration of location site.
+     */
+    exp: Schema.integer(),
```

Comment:
I do not understand exactly this question. This is related to the expiration that you want for the content claim you get back.
Also, probably better to not tie the implementation with this, this follows the spec, so I would recommend to discuss this at the spec level via issue/PR rather than here

Reply:
Idea here is that you tell service what do you want content location commitment expiration to be, which that will be encoded as expiry of the issued location claim. We should not infer desired expiry of commitment from the `blob/accept` invocation expiry because later tells time frame within which invocation can be exercised and not how long you want it's result to be valid for.

We could probably drop this for now as I think we decided that we default to non-expiring commitments. 

@vasco-santos we also should probably either make this field optional or allow non-expring commitments by allowing `null` here.

Reply:
Ok, I will change this to `ttl` and make it optional for now